### DISCUSSION TITLE
TagFilter Macro for 5.2+

### DISCUSSION THREAD

**Discussion by twoxfh** [TiddlyWiki/TiddlyWiki5#6984] [Q&A]
Hello! new TW5 user and I am finding new ways of working I wish I had explored years ago. I am a tag heavy user and my workflow typically drills down to specific content. 

Question:
@tobibeer created [TagFilter](http://tagfilter.tiddlyspot.com/#Start) for TW 5.1.7 in 2015, does anyone know of this functionality in the current TW 5.2.4? 

I did try to import the tagfilter macro in the TW 5.2.4 wiki and got some odd behavior which prevented its use. I really appreciate the dynamic nature of the macro to drilldown any tag levels, my own experiments have been predefined selects with limited depth. 

Thoughts?

**Comment by pmario** [comment]
The dependent PRs mentioned in the macro docs have never been implements. So the macro won't work. 

There is no formal description, what tagfilter actually does. As far as I understand it. 

- It takes a list of input tiddlers
- shows all tags those tiddlers have in a dropdown
- selects a title from the dropdown
- shows all tiddlers, that have the selected tags
- Only tiddlers which have all tags selected will be shown

Is that right?
Is that the functionality you actually want?

**Comment by twoxfh** [comment]
Correct as well as filters the selectable tags after each addition to only the remaining delta. Essentially keeps appending tag filter [tag[x]tag[y]...] Until no more can be selected.

I started a discussion on [talk](https://talk.tiddlywiki.org/t/tagfilter-like-functionality-for-tw-5-2-4/4819/3).tiddlywiki.org and since have found @bimlas locator plugin which so far works on 5.2.4. 

An out of box or native tag filter gui may be helpful for new or existing users who do not want the complexity of the locator plugin, however, I am finding it very useful.
